### PR TITLE
Drive v3 file snippet

### PR DESCRIPTION
**Pull request by RajeshGogo**
# Description
Drive-v3 file_snippets
export_pdf(Bug_id-232039687) : Download a document file in Pdf format.
move_file_to_folder(Bug_id-232040050) : Move one file to specified folder.
search_file(Bug_id-232177275) : Search a file in drive location
share_file(Bug_id-232177364) : Batch permission modification
upload_basic(Bug_id-232177235) : Insert a new file
touch_file(Bug_id-232177411) : Change file's modification timestamp

## Has it been tested?
- [X] Development testing done
- [ ] Unit or integration test implemented

## Checklist
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have performed a peer-reviewed with team member(s)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: drive/snippets/drive_v3/file_snippets/export_pdf.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ */
+// [START drive_export_pdf]
+
+/**
+ * Download a Document file in PDF format
+ * @param{string} realFileId file ID
+ * @return{obj} file status
+ * */
+async function exportPdf(realFileId) {
+  // Get credentials and build service
```
Comment (sqrrrl) [inline drive/snippets/drive_v3/file_snippets/export_pdf.js:24]: Move closer to statements so it reads better. Feels a little disconnected with the require statements in between

File: drive/snippets/drive_v3/file_snippets/export_pdf.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  // TODO (developer) - Use appropriate auth mechanism for your app
+
+  const {GoogleAuth} = require('google-auth-library');
+  const {google} = require('googleapis');
+
+  const auth = new GoogleAuth({scopes: 'https://www.googleapis.com/auth/drive'});
+  const service = google.drive({version: 'v3', auth});
+  fileId = realFileId;
```
Comment (sqrrrl) [inline drive/snippets/drive_v3/file_snippets/export_pdf.js:32]: fileId is undefined. Delete the assignment and just rename realFileId -> fileId.

See comments on #174 for additional details.

File: drive/snippets/drive_v3/file_snippets/move_file_to_folder.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ */
+// [START drive_move_file_to_folder]
+
+/**
+ * Move a file to specified folder
+ * @param{string} realFileId Id of the file to move
+ * @param{string} realFolderId Id of the folder to move
+ * @return{obj} file status
+ * */
+async function moveFileToFolder(realFileId, realFolderId) {
```
Comment (sqrrrl) [inline drive/snippets/drive_v3/file_snippets/move_file_to_folder.js:24]: Same comments as export_pdf -- align the auth comments & remove the fileId/folderId reassignments

File: drive/snippets/drive_v3/file_snippets/search_file.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+ * limitations under the License.
+ */
+// [START drive_search_file]
+
+/**
+ * Search file in drive location
+ * @return{obj} data file
+ * */
+async function searchFile() {
+  // Get credentials and build service
```
Comment (sqrrrl) [inline drive/snippets/drive_v3/file_snippets/search_file.js:23]: Same as other files

File: drive/snippets/drive_v3/file_snippets/search_file.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  // TODO (developer) - Use appropriate auth mechanism for your app
+
+  const {GoogleAuth} = require('google-auth-library');
+  const {google} = require('googleapis');
+
+  const auth = new GoogleAuth({scopes: 'https://www.googleapis.com/auth/drive'});
+  const service = google.drive({version: 'v3', auth});
+  const files = [];
+  const pageToken = null;
```
Comment (sqrrrl) [inline drive/snippets/drive_v3/file_snippets/search_file.js:32]: Not actually using page token here. Either don't page or add an outer loop to do paging:

```
let pageToken = null;
do {
  const res = await ....
  res.data.files.forEach(...);
  pageToken = res.data.nextPageToken;
} while (pageToken);
```

File: drive/snippets/drive_v3/file_snippets/share_file.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+/**
+ * Batch permission modification
+ * @param{string} realFileId file ID
+ * @param{string} realUser username
+ * @param{string} realDomain domain
+ * @return{obj} permission id
+ * */
+function shareFile(realFileId, realUser, realDomain) {
+  // Get credentials and build service
```
Comment (sqrrrl) [inline drive/snippets/drive_v3/file_snippets/share_file.js:26]: Same comments as other files re: auth comments & reassignments

File: drive/snippets/drive_v3/file_snippets/share_file.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+      'type': 'domain',
+      'role': 'writer',
+      'domain': 'example.com',
+    },
+  ];
+  permissions[0].emailaddress = realUser;
+  permissions[1].domain = realDomain;
+  // Using the NPM module 'async'
+  try {
+    async.eachSeries(permissions, function(permission) {
```
Comment (sqrrrl) [inline drive/snippets/drive_v3/file_snippets/share_file.js:53]: Was going to suggest using batch requests, but looks like the nodejs client doesn't support it :(

File: drive/snippets/drive_v3/file_snippets/share_file.js [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+      service.permissions.create({
+        resource: permission,
+        fileId: fileId,
+        fields: 'id',
+      })
+          .then(function(result) {
```
Comment (sqrrrl) [inline drive/snippets/drive_v3/file_snippets/share_file.js:59]: Indentation seems off. Reformat

File: drive/snippets/drive_v3/file_snippets/touch_file.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+ */
+// [START drive_touch_file]
+
+/**
+ * Change the file's modification timestamp.
+ * @param{string} realFileId ID of the file to change modified time
+ * @param{string} realTimestamp Timestamp to override Modified date time of the file
+ * @return{obj} modified Timestamp
+ **/
+async function touchFile(realFileId, realTimestamp) {
```
Comment (sqrrrl) [inline drive/snippets/drive_v3/file_snippets/touch_file.js:24]: Same comments as other files

File: drive/snippets/drive_v3/file_snippets/upload_basic.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+ * limitations under the License.
+ */
+// [START drive_upload_basic]
+
+/**
+ * Insert new file.
+ * @return{obj} file Id
+ * */
+async function uploadBasic() {
+  // Get credentials and build service
```
Comment (sqrrrl) [inline drive/snippets/drive_v3/file_snippets/upload_basic.js:23]: Same comments as other files

**Review by sqrrrl — CHANGES_REQUESTED** [review]

**Comment by anuraggoogler** [comment]
Done

**Pull request merged**